org.lflang.diagram/src/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.xtend

```diff
@@ -1078,8 +1082,31 @@ class LinguaFrancaSynthesis extends AbstractDiagramSynthesis<Model> {
 	
 	private def Iterable<KNode> createUserComments(EObject element, KNode targetNode) {
 		if (SHOW_USER_LABELS.booleanValue) {
-			val commentText = ASTUtils.findAnnotationInComments(element, "@label")
-			
+			var String commentText
+			// The list of elements where labels could be placed.
+			switch element {
+				Reaction : {
+					var list = element.getAttributes.filter[ it.attrName.toString === 'label' ]
+					if (list.length != 0)
+						commentText = list.get(0).getAttrParms.get(0).getValue.replaceAll("^\"|\"$", "")
+				}
+				Reactor : {
+					var list = element.getAttributes.filter[ it.attrName.toString === 'label' ]
+					if (list.length != 0)
+						commentText = list.get(0).getAttrParms.get(0).getValue.replaceAll("^\"|\"$", "")
+				}
+				Action : {
+					var list = element.getAttributes.filter[ it.attrName.toString === 'label' ]
+					if (list.length != 0)
+						commentText = list.get(0).getAttrParms.get(0).getValue.replaceAll("^\"|\"$", "")
+				}
+				Timer : {
+					var list = element.getAttributes.filter[ it.attrName.toString === 'label' ]
+					if (list.length != 0)
+						commentText = list.get(0).getAttrParms.get(0).getValue.replaceAll("^\"|\"$", "")
+				}
+			}
```

Comment:
Not sure how to resolve this code duplication... Xtend's support for polymorphism seems insufficient to avoid it.